src/github/activityBarViewProvider.ts

```diff
@@ -0,0 +1,363 @@
+		});
+	}
+
+	private async deleteBranch(message: IRequestMessage<any>) {
+		const branchInfo = await this._folderRepositoryManager.getBranchNameForPullRequest(this._item);
+		const actions: (vscode.QuickPickItem & { type: 'upstream' | 'local' | 'remote' })[] = [];
+
+		if (this._item.isResolved()) {
+			const branchHeadRef = this._item.head.ref;
+
+			actions.push({
+				label: `Delete remote branch ${this._item.remote.remoteName}/${branchHeadRef}`,
+				description: `${this._item.remote.normalizedHost}/${this._item.remote.owner}/${this._item.remote.repositoryName}`,
+				type: 'upstream',
+				picked: true
+			});
+		}
+
+		if (branchInfo) {
+			const preferredLocalBranchDeletionMethod = vscode.workspace.getConfiguration('githubPullRequests').get<boolean>('defaultDeletionMethod.selectLocalBranch');
+			actions.push({
+				label: `Delete local branch ${branchInfo.branch}`,
+				type: 'local',
+				picked: !!preferredLocalBranchDeletionMethod
+			});
+
+			const preferredRemoteDeletionMethod = vscode.workspace.getConfiguration('githubPullRequests').get<boolean>('defaultDeletionMethod.selectRemote');
+
+			if (branchInfo.remote && branchInfo.createdForPullRequest && !branchInfo.remoteInUse) {
+				actions.push({
+					label: `Delete remote ${branchInfo.remote}, which is no longer used by any other branch`,
+					type: 'remote',
+					picked: !!preferredRemoteDeletionMethod
+				});
+			}
+		}
+
+		if (!actions.length) {
+			vscode.window.showWarningMessage(`There is no longer an upstream or local branch for Pull Request #${this._item.number}`);
+			this._replyMessage(message, {
+				cancelled: true
+			});
+
+			return;
+		}
+
+		const selectedActions = await vscode.window.showQuickPick(actions, {
+			canPickMany: true,
+			ignoreFocusOut: true
+		});
+
+		if (selectedActions) {
+			const isBranchActive = this._item.equals(this._folderRepositoryManager.activePullRequest);
+
+			const promises = selectedActions.map(async (action) => {
+				switch (action.type) {
+					case 'upstream':
+						return this._folderRepositoryManager.deleteBranch(this._item);
+					case 'local':
+						if (isBranchActive) {
+							if (this._folderRepositoryManager.repository.state.workingTreeChanges.length) {
+								const response = await vscode.window.showWarningMessage(`Your local changes will be lost, do you want to continue?`, { modal: true }, 'Yes');
+								if (response === 'Yes') {
+									await vscode.commands.executeCommand('git.cleanAll');
+								} else {
+									return;
+								}
+							}
+							// await this._folderRepositoryManager.repository.checkout(this._repositoryDefaultBranch);
+						}
+						return await this._folderRepositoryManager.repository.deleteBranch(branchInfo!.branch, true);
+					case 'remote':
+						return this._folderRepositoryManager.repository.removeRemote(branchInfo!.remote!);
+				}
+			});
+
+			await Promise.all(promises);
+
+			// this.refreshPanel();
+			vscode.commands.executeCommand('pr.refreshList');
```

Comment:
Maybe during October we do a purge of all of these `executeCommand`s

Reply:
yes, definitely!